govcd/disk_test.go

```diff
@@ -321,6 +321,39 @@ func (vcd *TestVCD) Test_AttachedVMDisk(check *C) {
 	check.Assert(vmRef, NotNil)
 	check.Assert(vmRef.Name, Equals, vm.VM.Name)
 
+	// Detach disk
+	err = vcd.detachIndependentDisk(Disk{disk.Disk, &vcd.client.Client})
+	check.Assert(err, IsNil)
+}
+
+// Checks whether an independent disk is attached to a VM, and detaches it
+func (vcd *TestVCD) detachIndependentDisk(disk Disk) error {
+
+	// See if the disk is attached to the VM
+	vmRef, err := disk.AttachedVM()
+	if err != nil {
+		return err
+	}
+	// If the disk is attached to the VM, detach disk from the VM
+	if vmRef != nil {
+
+		vm, err := vcd.client.FindVMByHREF(vmRef.HREF)
+		if err != nil {
+			return err
+		}
+
+		// Detach the disk from VM
+		task, err := vm.DetachDisk(&types.DiskAttachOrDetachParams{
+			Disk: &types.Reference{
+				HREF: disk.Disk.HREF,
+			},
+		})
+		err = task.WaitTaskCompletion()
```

Comment:
Missing error handling from DetachDisk

Reply:
Done. Also added the same error handling in the cleanup procedure